sensortag20/BleSensorTag/src/main/java/com/example/ti/ble/sensortag/ButtonPressListener.java

```diff
@@ -91,8 +95,41 @@ public void buttonEvent(boolean event) {
         lastMilliseconds = currentMilliseconds;
     }
 
-    public Long[] onTimeout() {
+    public void onTimeout() {
         mTimer.reset();
-        return resetValues();
+        makePost();
+        resetValues();
+        return;
+    }
+
+    public Long[] getValues(){
+
+        // Still not sure why the first value is useless, but this works. ¯\_(ツ)_/¯
+        if (! mList.isEmpty() ) {
+            mList.remove(0);
+        }
+
+        Long valuesToSend[] = mList.toArray(new Long[mList.size()]);
+
+        return valuesToSend;
+    }
+
+    public void makePost(){
+
+        Long[] values = getValues();
+
+        Gson gson = new GsonBuilder().create();
+        String gSonData = gson.toJson(values);
+
+        try {
+            OkHttpClient client = new OkHttpClient();
+
+            RequestBody postData = new FormBody.Builder().add("data", gSonData).build();
+            Request request = new Request.Builder().url(Endpoint).post(postData).build();
+            Response response = client.newCall(request).execute();
+            System.out.println(response);
+        }catch(Exception e) {
+            System.exit(0);
```

Comment:
I'm concerned that this isn't the best way to handle an exception. It will work for now, but I don't think we should just kill the app if we get an exception.